packages/core/spec/share/asciidoctor-spec.js

```diff
@@ -1232,6 +1232,31 @@ April,32
 `)
       })
     })
+
+    describe('Syntax Highligther', function () {
+      it('should return the syntax highlighter class registered for the specified name', () => {
+        const highlightjsSyntaxHighlighter = asciidoctor.SyntaxHighlighter.for('highlight.js')
+        expect(highlightjsSyntaxHighlighter.$$name).to.equal('HighlightJsAdapter')
+        const rougeSyntaxHighlighter = asciidoctor.SyntaxHighlighter.for('rouge')
+        expect(rougeSyntaxHighlighter).to.be.undefined()
+      })
+
+      it('should create a new syntax highlighter', () => {
+        asciidoctor.SyntaxHighlighterBase.create('unavailable', function () {
```

Comment:
My goal was to be as close as: https://github.com/asciidoctor/asciidoctor/blob/7d29276eb1db617ade8055f119b86db002d6f4b6/test/syntax_highlighter_test.rb#L66

For some reason, Asciidoctor.js is throwing the following exception if the name is not defined: `asciidoctor: FAILED: <stdin>: Failed to load AsciiDoc document - unavailable must specify a value for `name'`.

Reply:
I don't really like the idea that create is even being used in this context. The create method on the base class is definitely for use by the processor only.

Reply:
Ok